Bosnia

Vesihiisi opened the pull request:
Mapping: https://www.wikidata.org/w/index.php?title=User:Alicia_Fagerving_(WMSE)/Bosnia
Preview: https://www.wikidata.org/wiki/User:Alicia_Fagerving_(WMSE)/Bosnia_prev

Task: https://phabricator.wikimedia.org/T179828

lokal-profil commented:
Could you put together something like [https://www.wikidata.org/wiki/User:Eric_Luth_(WMSE)/Namibia](this matching table) to give an easy mapping overview from the source data columns to properties?

Vesihiisi commented:
Added a table here https://www.wikidata.org/w/index.php?title=User:Alicia_Fagerving_(WMSE)/Bosnia

lokal-profil commented on importer/Bosnia.py:
```diff
@@ -0,0 +1,321 @@
+        quals = {}
+        if date_parsed:
+            date_dict = utils.datetime_object_to_dict(date_parsed)
+            quals["start_time"] = {"time_value": date_dict}
```
utils.package_time(date_dict)

Vesihiisi replied:
:white_check_mark:

lokal-profil commented on importer/Bosnia.py:
```diff
@@ -0,0 +1,321 @@
+        else:
+            to_add = default
+
+        self.add_statement("is", to_add, refs=self.source)
```
add `, if_empty=True)` to not add when there are pre-existing P31s

Vesihiisi replied:
:white_check_mark:

lokal-profil commented on importer/Bosnia.py:
```diff
@@ -0,0 +1,321 @@
+                    self.add_statement(
+                        "coordinates", coord_tuple, refs=self.source)
+                except ValueError:
+                    print("Nope coords: {}".format(self.coordinates))
```
not add to report?

Vesihiisi replied:
:white_check_mark:

lokal-profil commented on importer/Bosnia.py:
```diff
@@ -0,0 +1,321 @@
+        self.update_labels()
+
+
+def generate_filenames(tablename, timestamp):
```
why not just `import importer` in the start then call `importer.make_filenames` instead?

lokal-profil replied:
True for several of the other functions as well and would make it easier to see which parts of the code are new and which are re-used.

Vesihiisi replied:
Because at that point I had no way of knowing whether I'd be able to use the importer functions or whether and to what degree they'd have to be modified.

lokal-profil commented on importer/Bosnia.py:
```diff
@@ -0,0 +1,321 @@
+    """Load the files with mappings of various values."""
+    mapping_files = {}
+    mapping_files["_static"] = {}
+    files_to_get = ["ba_(bs).json", "bosnia_adm.json", "bosnia_types.json"]
```
`ba_(bs).json` is missing from the patch

Vesihiisi replied:
:white_check_mark:

lokal-profil commented on importer/Bosnia.py:
```diff
@@ -0,0 +1,321 @@
+        last_edit = "2017-09-08"
+        timestamp = utils.date_to_dict(last_edit, "%Y-%m-%d")
+        prop_stated = self.props["stated_in"]
+        prop_date = self.props["publication_date"]
```
Should this actually be "retrieval date" since the google doc isn't the actual date of publication?

Vesihiisi replied:
:white_check_mark: this actually required hacking the Uploader.... :open_mouth:

Vesihiisi replied:
...though of course we don't _really_ know how accurate this date is, regardless of what prop we stick on it....

lokal-profil commented:
Could we use the number in the url (just before the `?`) as an WLM identifier with a `BA-` prefix?

Vesihiisi commented:
> Could we use the number in the url (just before the `?`) as an WLM identifier with a `BA-` prefix?

Some of the items don't have url's (they haven't been uploaded to the system according to the notes). So the coverage of the WLM ID might be misleading as to the number of objects we've uploaded.

Vesihiisi commented:
preview up to here.

lokal-profil commented:
>> Could we use the number in the url (just before the `?`) as an WLM identifier with a `BA-` prefix?

> Some of the items don't have url's (they haven't been uploaded to the system according to the notes). So the coverage of the WLM ID might be misleading as to the number of objects we've uploaded. This also affects the finding of existing matches... There really isn't a good solution apart from not uploading the items without url's, which is not a good solution.

So what id are we adding today? Fro the preview it looks like we are not adding any, meaning we also cannot detect existing matches nor ensure that we don't upload the same item twice... Or am I misreading it?

lokal-profil commented:
Only looks like 6 out of the 170 don't have a url

Vesihiisi commented:
> So what id are we adding today? Fro the preview it looks like we are not adding any, meaning we also cannot detect existing matches nor ensure that we don't upload the same item twice... Or am I misreading it?

Yeah, there's no ID included right now and no, I don't like it either :/

Vesihiisi commented:
So I guess the extracted ID's from the URL's should be used as WLM ID, but what do with the non-urled ones?

lokal-profil commented:
>So I guess the extracted ID's from the URL's should be used as WLM ID, but what do with the non-urled ones?

Either skip them (for now) or assign them a differently structured identitfier (which we then also add to the document). E.g. `BA-1234` for url ones and `BA-wiki-01` for the others. Ideally we would propose a Property and then migrate the first type at some future point.

Vesihiisi commented:
Added WLM id from URL and skipped the ones without URL's... only 6 items as far as i can see.

Vesihiisi commented:
Preview updated here.

lokal-profil commented on importer/Bosnia.py:
```diff
@@ -197,6 +212,7 @@ def construct_wd_item(self, mapping, data_files=None):
         self.set_street_address()
         self.update_descriptions()
         self.update_labels()
+        # self.set_wd_item(self.find_matching_wikidata(mapping))
```
Why is this commented out?

Vesihiisi replied:
....because i forgot about it

lokal-profil commented on importer/Bosnia.py:
```diff
@@ -98,20 +97,25 @@ def set_is(self):
-                                   refs=self.source)
+                return self.url.split("?")[0]
+
+    def set_described(self):
```
Do we actually want this as a qualifier on heritage status?

Vesihiisi replied:
This is not being added as a qualifier on heritage status.

Vesihiisi replied:
oh god now i get it

Vesihiisi commented:
Moved the url to quals of heritage status, updated preview, [pushed an item to the Sandbox](https://www.wikidata.org/w/index.php?title=Q4115189&oldid=590235063).
